app/js/questionNav.js

```diff
@@ -50,7 +52,22 @@ function prev() {
         document.querySelector(".active").classList.remove("active")
         document.querySelector(".q_" + current).classList.add("active")
     }
+    updateFlagStatus()
+}
+
+function updateFlagStatus() {
+    let qid  = document.querySelector('#questions .question.active').dataset.questionid
+    document.querySelector('#flag-checkbox').checked = flaggedQuestions[qid]
+
+    //('#questions-nav') is a placeholder for the page's number on the navbar
+    if(flaggedQuestions[qid]) {
+        document.querySelector('#questions-nav').classList.add('glyphicon','glyphicon-flag')
+    } else {
+        document.querySelector('#questions-nav').classList.remove('glyphicon','glyphicon-flag')
+    }
```

Comment:
When this gets merged into the main branch the flag will have to be inside an element inside the nav element.